<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <arg name="track_serial_no"        default="845412110931"/>
    <arg name="track_camera"           default="track"/>

    <arg name="depth_over_serial_no"  default="838212070287"/>
    <arg name="depth_over_camera"      default="cam_1"/>

    <arg name="depth_under_serial_no"   default="818312071183"/>
    <arg name="depth_under_camera"     default="cam_2"/>

    <arg name="lis_header_id"       default="track_odom_frame" />
    <arg name="lis_child_id"        default="track_pose_frame" />
    <arg name="pub_header_id"       default="track_pose_frame" />
    <arg name="pub_child_id"        default="$(arg depth_over_camera)_link" />

    <arg name="cam_over"            default="$(arg depth_over_camera)_link" />
    <arg name="cam_under"           default="$(arg depth_under_camera)_link" />
    <arg name="x"                   default="-0.045" />
    <arg name="y"                   default="0.0" />
    <arg name="z"                   default="-0.045" />
    <arg name="yaw"                 default="0" />
    <arg name="pitch"               default="40" />
    <arg name="roll"                default="0" />


    <include file="$(find realsense2_camera)/launch/rs_t265Name.launch" >
        <arg name="serial_no"   value="$(arg track_serial_no)"/>
        <arg name="camera"      value="$(arg track_camera)"/>
    </include>

    <include file="$(find realsense2_camera)/launch/rs_camera1.launch" >
        <arg name="serial_no"   value="$(arg depth_over_serial_no)"/>
        <arg name="camera"      value="$(arg depth_over_camera)"/>
    </include>
    
    <include file="$(find realsense2_camera)/launch/rs_camera2.launch" >
        <arg name="serial_no"   value="$(arg depth_under_serial_no)"/>
        <arg name="camera"      value="$(arg depth_under_camera)"/>
    </include>

    <include file="$(find pc_tf)/launch/pc_transform.launch" >
        <arg name="lis_header_id"   value="$(arg lis_header_id)" /> 
        <arg name="lis_child_id"    value="$(arg lis_child_id)" /> 
        <arg name="pub_header_id"   value="$(arg pub_header_id)" /> 
        <arg name="pub_child_id"    value="$(arg pub_child_id)" /> 
    </include>

    <node pkg="realsense2_camera"
        name="my_static_tf2_broadcaster" 
	    type="cams_tf.py" >
        <param name="from_cam" value="$(arg cam_over)" /> 
        <param name="to_cam" value="$(arg cam_under)" /> 
        <param name="x" value="$(arg x)" /> 
        <param name="y" value="$(arg y)" /> 
        <param name="z" value="$(arg z)" /> 
        <param name="yaw" value="$(arg yaw)" /> 
        <param name="pitch" value="$(arg pitch)" /> 
        <param name="roll" value="$(arg roll)" /> 
    </node>

</launch>
